### PR TITLE
[docs] Improve documentation of normalization layers

### LINKED ISSUE #51455: [docs] Improve documentation for LayerNorm
https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html: `The mean and standard-deviation are calculated separately over the last certain number dimensions which have to be of the shape specified by normalized_shape`

What's not clear from this excerpt (and the rest of the text):

1. What dimensions is the mean computed over? What is the shape of the computed mean/std? If it's the last few dimensions as suggested by this text, then does it mean that for BCHW layout, height and width must be fixed and passed at LayerNorm construction time? This seems wrong. Also expression "calculated separately over the last certain number of dims" may imply exactly the reverse: that the first dims are aggregated over. What seems to be happening is that "mean and standard-deviation are calculated over all except the first dimension (assumed batch dimension) and the specified last dimension (assumed channels dimensions)".
2. If the last few dimensions (including spatial/temporal) are aggregated over, why does LayerNorm need their (potentially dynamic) shape at construction time?

2. Is there any layout are assumed for a typical usecase (with fixed number of channels and various time/spatial dimension)? BTC? BCT? This is important, because PyTorch usually uses BC... logical layouts in other contexts

3. Could it include a visual explanation of what dimensions are aggregated over?

The examples given in the docs don't include dimension semantics (such as Channels, Time, Height, Width), so I couldn't extract any information from them.

E.g. an example of type:
```python
import torch

ln = torch.nn.LayerNorm(256, elementwise_affine = False)

# BTC
a = torch.randn(8, 1000, 256) * 100 + 200
print(a[0, ...].mean(), a[0, ...].std())
#tensor(199.9173) tensor(99.8517)

print(a[0, 0].mean(), a[0, 0].std())
#tensor(206.8604) tensor(104.7011)

b = ln(a)
print(b[0, ...].mean(), b[0, ...].std())
# tensor(2.2389e-09) tensor(1.0000)
print(a[0, 0].mean(), a[0, 0].std())
#tensor(-2.6077e-08) tensor(1.0020)

# what's going on?
# Does LayerNorm standardize every (b, ...) slice as illustrated in GroupNorm paper's figure?
# Does LayerNorm standardizes every (b, t, :) slice (i.e. standardizes embedding vectos themselves - does it even make sense)?
```
is much clearer.

Phrasing "over the last certain number dimensions" seems very ambiguous, as the aggregation seems to happen actually over not specified dimensions.

cc @jlin27 @mruberry @albanD @jbschlosser

### LINKED ISSUE #75862: LayerNorm and GroupNorm with num_groups=1 not equivalent
### 🐛 Describe the bug

`LinearNorm` and `GroupNorm` with `num_groups=1` should be equivalent but they are not

```python
from torch import nn
import torch
import random
random.seed(0)
torch.manual_seed(0)
x = torch.randn((1, 8, 2, 2))


x_norm = x.permute(0, 2, 3, 1)
x_norm = nn.LayerNorm(8, eps=1e-6)(x_norm)
x_norm.permute(0, 3, 1, 2)

print(x_norm[0,:2,:2,:2])
```

```
tensor([[[-1.4655,  0.7272],
         [-0.6575,  0.8390]],

        [[-0.8825, -1.0009],
         [-0.3389, -1.8741]]], grad_fn=<SliceBackward0>)
```

```python
x_norm = nn.GroupNorm(1, 8, eps=1e-6)(x)

print(x_norm[0,:2,:2,:2])
```

```
tensor([[[-1.1255, -1.1518],
         [-0.2556, -0.4378]],

        [[ 0.8369,  0.6812],
         [-0.3206, -2.1088]]], grad_fn=<SliceBackward0>)
````

### Versions

```
Collecting environment information...
PyTorch version: 1.11.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.13.0-39-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1080 Ti
Nvidia driver version: 510.54
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.3
[pip3] pytorch-lightning==1.5.10
[pip3] pytorch-pretrained-biggan==0.1.1
[pip3] torch==1.11.0
[pip3] torchaudio==0.11.0
[pip3] torchcam==0.3.1
[pip3] torchinfo==1.6.5
[pip3] torchmetrics==0.7.3
[pip3] torchvision==0.12.0
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               h2bc3f7f_2  
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py39h7f8727e_0  
[conda] mkl_fft                   1.3.1            py39hd3c417c_0  
[conda] mkl_random                1.2.2            py39h51133e4_0  
[conda] mypy-extensions           0.4.3                    pypi_0    pypi
[conda] numpy                     1.22.3                   pypi_0    pypi
[conda] pytorch-lightning         1.5.10                   pypi_0    pypi
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] pytorch-pretrained-biggan 0.1.1                    pypi_0    pypi
[conda] torch                     1.11.0                   pypi_0    pypi
[conda] torchaudio                0.11.0                   pypi_0    pypi
[conda] torchcam                  0.3.1                    pypi_0    pypi
[conda] torchinfo                 1.6.5                    pypi_0    pypi
[conda] torchmetrics              0.7.3                    pypi_0    pypi
[conda] torchvision               0.12.0                   pypi_0    pypi
```

### PR DESCRIPTION
**Pull request by AlexeyKruglov**
Fixes #51455 (its documentation part, not the implementations; the impl part of that issue seems to be covered by #113750 )

Fixes #75862

This PR improves documentation for normalization layers.

Changes:
1. Clarify normalization layers' documentation by explicitly listing the remaining dimension indices of tensors after the reductions for means/variances and the list of indices for the learnable transformations using a visual but formal mathematical notation, that allows one to quickly find which tensor depends on which indices. Previously this information was scattered through the text and was sometimes ambiguous, which could confuse users. Layers covered: LayerNorm, GroupNorm, RMSNorm, InstanceNorm1d, InstanceNorm2d, InstanceNorm3d, BatchNorm1d, BatchNorm2d, BatchNorm3d, SyncBatchNorm.
2. Fix LocalResponseNorm doc to match its implementation and the input/output notation used with the other layers.
3. Remove/fix misleading statements in GroupNorm docs (equivalence to LayerNorm when num_gropus=1), RMSNorm (affine->scaling), BatchNorm2d and BatchNorm3d.
4. Fix few typos in the docstrings.

### CODE REVIEW

**Comment by AlexeyKruglov** [comment]
@mikaylagawarecki @jbschlosser @albanD Please enable tests for me — this is my first PR here. And a kind reminder about the review. 
This PR contains only documentation changes, no code changes (including examples), compiled docs on my PC to check, and rebased to viable/strict.

**Comment by albanD** [comment]
@claude review this please

**Comment by AlexeyKruglov** [comment]
@albanD 

0. About C/G agree, fixed.

1. I think in this case it's better to keep the explanation of the mathematical side of the op to the equations part and input/output format to the args part. Since there's no common notation for optional function arguments, using "[,t]" or ",t?" may cause more confusion with the uncommon notation than help with the input format. Math part is here to explain the mathematical side, and we don't have to duplicate in/out details in both places unless there is a good clear way to do it, IMO.

2. P.S.: Rewrapping line — agreed & fixed. <del>There's a short line ending with "are set" because [CONTRIBUTING.md](https://github.com/pytorch/pytorch/blob/main/CONTRIBUTING.md#writing-documentation) requires limiting docstring lines to 80 characters (counting from column 5, as I understand it). The existing lines are longer than 80, so I rewrapped the lines near my changes, and this is the last line of the rewrapped part. I didn't rewrap whole paragraphs to make it easier to spot changes during the review or in git blame later. I also considered/tried wrapping it to less that 80 characters to avoid having a short line at the end, but decided to go with full 80 characters so that the editors after me could do the same without the need to rewrap my part.</del>

3. ``RMSNorm`` is done differently because (in the end) it contains an explicit expression for RMS. I considered several options of writing it down, and this one seems the least potentially confusing. Unlike the other layers, it contains both "i_0, ..., i_{D-d-1}" and "i_0, ..., i_{D-1}", and using "n, ..., i_{D-d-1}" might lead to confusion whether n is i_0 or i_1. Using just "n, ..." with an underbrace would require the underbrace to be repeated too often because sometimes there'd be (D-d) and sometimes D terms.

4. Agree about ``\phantom`` that its rendering needs to be checked in the docs preview. I checked it on my PC. But since I'm a [first-time contributor, the reviewer "may need to" approve running the test jobs](https://github.com/pytorch/pytorch/wiki/Typical-Pull-Request-Workflow#review) for me to compile the docs. At least, they're still not compiled yet for whatever reason (the pytorch-bot's link to the docs above leads to an XML with "The specified key does not exist" error).

If you think it's still preferrable to fix 1 or 3 or see a better option of fixing, please let me know.

**Comment by AlexeyKruglov** [comment]
@albanD @jbschlosser @mikaylagawarecki ping. Please review my PR.

**Comment by AlexeyKruglov** [comment]
@jerryzh168 ping. Please review my PR. It's been 3+ weeks since I updated my PR after the last review.

**Comment by AlexeyKruglov** [comment]
@albanD @jbschlosser @mikaylagawarecki ping. Please review my PR. Waiting for 2 months already.
